src/pages/VeSDL/index.tsx

```diff
@@ -10,48 +10,228 @@ import {
+    try {
+      if (sdlTokenValue.gt(Zero)) {
+        await checkAndApproveTokenForTrade(
+          sdlContract,
+          votingEscrowContract.address,
+          account,
+          sdlTokenValue,
+          infiniteApproval,
+        )
+      }
+
+      if (shouldCreateLock && unlockTimeStamp) {
+        const txn = await votingEscrowContract?.create_lock(
+          sdlTokenValue,
+          unlockTimeStamp,
+        )
+        void enqueuePromiseToast(chainId, txn.wait(), "createLock")
+      } else if (shouldIncreaseAmoutAndLockEnd) {
+        const txnIncreaseAmount = await votingEscrowContract.increase_amount(
+          sdlTokenValue,
+        )
+        await txnIncreaseAmount.wait()
+        const txnIncreaseUnlockTime =
+          await votingEscrowContract.increase_unlock_time(
+            BigNumber.from(unlockTimeStamp),
+          )
+        await txnIncreaseUnlockTime.wait()
+        enqueueToast("success", "Increased amount and unlock time")
+      } else if (shouldIncreaseAmount) {
+        // Deposit additional SDL into and existing lock
+        const txn = await votingEscrowContract.increase_amount(sdlTokenValue)
+        await txn.wait()
+        void enqueuePromiseToast(chainId, txn.wait(), "increaseLockAmt")
+      } else if (shouldIncreaseLockEnd) {
+        // Extend the unlock time on a lock that already exists
+        const txn = await votingEscrowContract.increase_unlock_time(
+          BigNumber.from(unlockTimeStamp),
+        )
+        await txn.wait()
+        void enqueuePromiseToast(chainId, txn.wait(), "increaseLockEndTime")
+      }
+      void fetchData()
+      resetFormState()
+    } catch (err) {
+      console.log(err)
+    }
   }
 
-  const prevUnlockDate: Date = new Date("2022-5-23")
-  const addLockMos = date ? differenceInMonths(date, prevUnlockDate) : null
+  const handleUnlock = async () => {
+    if (votingEscrowContract && chainId) {
+      const txn = await votingEscrowContract?.force_withdraw()
+      void enqueuePromiseToast(chainId, txn.wait(), "unlock")
+      void fetchData()
+    }
+  }
+
+  const addLockMos =
+    unlockDate && !isNaN(unlockDate.valueOf())
+      ? differenceInMonths(unlockDate, lockEnd || new Date())
+      : 0
 
   const lockHelperText = () => {
-    const sdlTokenValue = parseEther(sdlTokenRawVal.trim() || "0")
-    if (sdlTokenValue.gt(Zero) && !addLockMos) {
-      return t("increaseLockAmount", { addLockAmt: sdlTokenRawVal })
-    } else if (sdlTokenValue.gt(Zero) && addLockMos && addLockMos > 0) {
-      return t("increaseLockAmountAndTime", {
-        addLockMos,
-        addLockAmt: sdlTokenRawVal,
-      })
-    } else if (sdlTokenValue.eq(Zero) && addLockMos && addLockMos > 0) {
-      return t("increaseLockTime", { addLockMos })
+    if (lockedSDLVal.isZero()) {
+      if (sdlTokenValue.gt(Zero) && addLockMos > 0)
+        return t("lockSdl", {
+          sdlAmount: sdlToken.sdlTokenInputVal,
+          period: addLockMos,
```

Comment:
Do we need to format these months into years?

Reply:
@hammeiam 
Do you want to show the date distance as the format of year, month, week? EX: 1year 2 month 2 week

Reply:
Yes you're correct. Are we currently showing something like `Locking SDL for 200 months`?